source/extensions/access_loggers/stderr/config.cc

```diff
@@ -0,0 +1,63 @@
+#include "extensions/access_loggers/stderr/config.h"
+
+#include <memory>
+
+#include "envoy/extensions/access_loggers/stderr/v3/stderr.pb.h"
+#include "envoy/extensions/access_loggers/stderr/v3/stderr.pb.validate.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/common/logger.h"
+#include "common/config/utility.h"
+#include "common/formatter/substitution_format_string.h"
+#include "common/formatter/substitution_formatter.h"
+#include "common/protobuf/protobuf.h"
+
+#include "extensions/access_loggers/common/file_access_log_impl.h"
+#include "extensions/access_loggers/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace File {
+
+AccessLog::InstanceSharedPtr
+StderrAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
+                                                AccessLog::FilterPtr&& filter,
+                                                Server::Configuration::FactoryContext& context) {
+  const auto& fal_config = MessageUtil::downcastAndValidate<
+      const envoy::extensions::access_loggers::stderr::v3::StdErrorAccessLog&>(
+      config, context.messageValidationVisitor());
+  Formatter::FormatterPtr formatter;
+  if (fal_config.access_log_format_case() ==
+      envoy::extensions::access_loggers::stderr::v3::StdErrorAccessLog::AccessLogFormatCase::
+          kLogFormat) {
+    formatter = Formatter::SubstitutionFormatStringUtils::fromProtoConfig(fal_config.log_format(),
+                                                                          context.api());
+  } else if (fal_config.access_log_format_case() ==
+             envoy::extensions::access_loggers::stderr::v3::StdErrorAccessLog::AccessLogFormatCase::
+                 ACCESS_LOG_FORMAT_NOT_SET) {
+    formatter = Formatter::SubstitutionFormatUtils::defaultSubstitutionFormatter();
+  }
```

Comment:
This is mostly copied. Can we share this in a common base class, template, etc.?

Reply:
It also bothers me that they are the same but the types were different because of the enums in the protos, I will look into it more.